### ISSUE TITLE
Bug: Chore & Battery tracking dropdowns clears after losing focus

### ISSUE THREAD

**Issue by akoshpinter** [grocy/grocy#1560]
Hi,

When entering a non _grocycode_ entry in the dropdown, i.e. searching by name and selecting from the suggestion list, the entry will be lost as soon as the focus is moved to another field, making it impossible to track a chore or battery that happened in the past. 

Seems there was an error in the squashed commit: 91d8eaeb7433c0817abf332c1a81dc4a4b9bdc53 from 13.07.21, where the logic always clears the inputs even if it was not a _grocycode_, because the **possibleOptionElement** will always be empty for non _grocycode_ entries. 

Version affected: 3.1.0

**Comment by berrnd** [comment]
Seems to be only a problem when selecting the entry via mouse or touch, not when typing + `TAB` or `ENTER`.

**Comment by akoshpinter** [comment]
Yes, it definitely happened on the phone and clicking with the mouse. The `TAB` or `ENTER` works because the dropdown menu is still opened when the `blur` event happens, so the affected code is not executed. 😃
